gridt/src/app/core/settings.service.ts

```diff
@@ -30,15 +30,15 @@ export class SettingsService {
    * @param identity Settings to be stored.
    */
   public setLocalIdentity$(identity: Identity): Observable<boolean> {
-    console.debug(`Storing identity: ${JSON.stringify(identity)} into the local storage.`);
-    let loggedIn: boolean;
-    this.auth.isLoggedIn$.subscribe( (val) => {loggedIn = val;});
-
-    if (loggedIn) {
-      this.secStore.set$("identity", identity).subscribe();
-      return of(true);
-    } else {
-      return throwError("Not logged in: can't store identity in local storage.");
-    }
+    return this.auth.isLoggedIn$.pipe(
+      take(1),
+      tap( (val) => {
+        if (val === true) {
```

Comment:
This should be a filter 

https://www.learnrxjs.io/learn-rxjs/operators/filtering/filter

Reply:
I'm not sure about that. If you use a filter as you say then upon a failed auth, it will just do nothing at all. While I would like it to then throw an error. Hence I used tap, such that any value will be passed to the map.

Reply:
Fair, throwing away the error would not be a good idea. You could use flatmap to throw an error on the false condition and true on the other.